packages/datagateway-download/src/downloadStatus/downloadProgressIndicator.component.tsx

```diff
@@ -0,0 +1,66 @@
+import { Box, LinearProgress, Typography } from '@mui/material';
+import React from 'react';
+import {
+  useDownload,
+  useDownloadPercentageComplete,
+} from '../downloadApiHooks';
+import { useTranslation } from 'react-i18next';
+
+interface DownloadProgressIndicatorProps {
+  downloadId?: number;
+}
+
+/**
+ * A component for showing the download progress of the given download item.
+ * @param download The download item that this component should show the progress of.
+ * @constructor
+ */
+function DownloadProgressIndicator({
+  downloadId,
+}: DownloadProgressIndicatorProps): JSX.Element {
+  const [t] = useTranslation();
+  const { data: download, isLoading: isLoadingDownload } = useDownload({
```

Comment:
Why do we requery for the download here? Could we not just pass the required props from the status tables to save us requerying for all the downloads in the table?

Reply:
The problem is that the data in the table are formatted (`FormattedDownload`), which makes it awkward to work on the status of the download. In theory I can use a combination of `t` from useTranslation and translation key and use it to check the download status, but it feels wrong.

Reply:
It feels like that would be more efficient than sending out more queries though right? Alternatively, we could do something like not use a select function to transform the downloads and instead do it in a separate variable (like what happens in `adminDownloadStatusTable` and then do something like this in the `cellContentRenderer`:

```
cellContentRenderer: ({
   rowData,
}: TableCellProps) => (
  <DownloadProgressIndicator
    download={data?.pages.flatMap((page) => page).find((download) => download.id === (rowData as FormattedDownload).id)}
  />
)
```

Also, couldn't you just make a hook that does the reverse of `formatDownload` - i.e. to transform from `FormattedDownload` to `Download` which you can use to abstract the `useTranslation` stuff and use the hook in `downloadProgressIndicator`?

Reply:
Yeah, when I was writing that piece of code I was thinking calling `useDownload` there should have been a cache hit because `DownloadConfimDialog` also calls it, but then I just realized the table could've been loaded before the dialog (so obvious but I missed it).

What about a map that maps download IDs to `Download`, so that we can just look up the map for the unformatted `Download` when it is needed?

Reply:
I don't see why the reverse of `formatDownload` is awkward, as we define both sides of the equation (theoretically there could be weird issues if the syadmin had duplicate string values in the translations but that's a niche concern I think). If you think having a map is more useful then go for it. A third alternative is to create a new field in `FormattedDownload` that's like `systemStatus` which keeps the original form of the status enum? For usages like this?

Reply:
> theoretically there could be weird issues if the syadmin had duplicate string values in the translations but that's a niche concern I think
That is exactly what I am thinking about! What if the status values map to the same user-facing label? But if it is not a concern, then it will be way easier to just do a reverse of `formatDownload`.

Reply:
I think I prefer adding an extra field to `FormattedDownload` - I was thinking adding a field called `formattedStatus` that stores the, well, formatted status, and the `status` will store the original value.

Reply:
I can do the same for `isDeleted`.